pithos/plugins/dbus_util/GioNotify.py

```diff
@@ -158,20 +163,58 @@ def set_hint(self, key, value):
             self._hints[key] = value
 
     def do_g_signal(self, sender_name, signal_name, parameters):
-        id, signal_value = parameters.unpack()
-        # We only care about our notifications.
-        if id != self._replace_id:
-            return
-        # In GNOME Shell at least this stops multiple
-        # redundant 'NotificationClosed' signals from being emmitted.
-        if (id, signal_name) == self._last_signal:
-            return
-        self._last_signal = id, signal_name
-        if signal_name == 'ActionInvoked':
-            self.emit('action-invoked', signal_value)
-            self._callbacks[signal_value]()
+        try:
+            id, signal_value = parameters.unpack()
```

Comment:
I know this was used before but can you avoid overriding `id` which is a builtin function.

Reply:
No problem